### DISCUSSION TITLE
How to enable warnings on unused variables?

### DISCUSSION THREAD

**Discussion by jborbik** [clangd/clangd#1533] [Q&A] [answered]
Same as title. How to enable warnings on unused variables? The [configuration](https://clangd.llvm.org/config.html) does not mention the unused variables and such warnings are not enabled by default.

**Comment by HighCommander4** [comment, marked as answer]
Clangd just passes through the diagnostics produced by the [clang frontend](https://clang.llvm.org/docs/DiagnosticsReference.html) and [clang-tidy](https://clang.llvm.org/extra/clang-tidy/).

[Unused variable warnings](https://clang.llvm.org/docs/DiagnosticsReference.html#wunused-variable) are produced by the clang frontend and enabled with the flag `-Wunused-variable`.

If you want these warnings to also show up when building, you would add this flag in your build system metadata (for example, `makefile` or `CMakeLists.txt`), and then re-generate the `compile_commands.json` for clangd to pick it up.

If you want these warnings shown in the editor but _not_ when building, you could specify the flag in a [clangd config file](https://clangd.llvm.org/config.html#add), for example:

```yaml
CompileFlags:
  Add: [-Wunused-variable]
```

**Reply by jborbik** [reply]
Thank you so much! That’s exactly what I was looking for :)

**Reply by rudfol** [reply]
ty so much that helped me too
